Assets/Mirror/Runtime/QuaternionReadWrite.cs

```diff
@@ -0,0 +1,348 @@
+            // removes largestIndex from a 
+            a &= ~(3u << 22);
+
+            const int bitLength = 22;
+
+            float x = scaleFromUInt(a, bitLength);
+            float y = scaleFromUInt(b, bitLength);
+            float z = scaleFromUInt(c, bitLength);
+
+            Vector3 small = new Vector3(x, y, z);
+            result = FromSmallerDimensions(largestIndex, small);
+            return result;
+        }
+
+        static void WriteQuaternionHalf(NetworkWriter writer, int largestIndex, Vector3 small)
+        {
+            // 22 bits
+            int bitLength = 9;
+            uint a = scaleToUInt(small.x, bitLength);
+            uint b = scaleToUInt(small.y, bitLength);
+            uint c = scaleToUInt(small.z, bitLength);
+
+            // split abc into 8+1 bits, pack extra 1 bit with largestIndex
+            uint a1 = byte.MaxValue & a;
+            uint a2 = byte.MaxValue & (a >> 8);
+
+            uint b1 = byte.MaxValue & b;
+            uint b2 = byte.MaxValue & (b >> 8);
+
+            uint c1 = byte.MaxValue & c;
+            uint c2 = byte.MaxValue & (c >> 8);
+
+            uint extra = ((uint)largestIndex) + (a2 << 2) + (b2 << 3) + (c2 << 4);
+
+            writer.WriteByte((byte)a1);
+            writer.WriteByte((byte)b1);
+            writer.WriteByte((byte)c1);
+            writer.WriteByte((byte)extra);
+        }
+        static Quaternion ReadQuaternionHalf(NetworkReader reader)
+        {
+            Quaternion result;
+            uint a1 = reader.ReadByte();
+            uint b1 = reader.ReadByte();
+            uint c1 = reader.ReadByte();
+            byte extra = reader.ReadByte();
+
+            // first 2 bytes
+            uint largestIndex = (uint)(extra & 3);
+
+            // get nth bit, then move to start
+            uint a2 = (uint)((extra & (1 << 2)) >> 2);
+            uint b2 = (uint)((extra & (1 << 3)) >> 3);
+            uint c2 = (uint)((extra & (1 << 4)) >> 4);
+
+            uint a = a1 | (a2 << 8);
+            uint b = b1 | (b2 << 8);
+            uint c = c1 | (c2 << 8);
+
+            const int bitLength = 9;
+
+            float x = scaleFromUInt(a, bitLength);
+            float y = scaleFromUInt(b, bitLength);
+            float z = scaleFromUInt(c, bitLength);
+
+            Vector3 small = new Vector3(x, y, z);
+            result = FromSmallerDimensions(largestIndex, small);
+            return result;
+        }
+
+
+        internal static int findLargestIndex(Quaternion q)
+        {
+            int index = 0;
+            float current = Mathf.Abs(q.x);
+
+            for (int i = 1; i < 4; i++)
+            {
+                float next = Mathf.Abs(q[i]);
+                if (next > current)
+                {
+                    index = i;
+                    current = next;
+                }
+            }
+
+            return index;
+        }
+
+
+        static Vector3 getSmallerDimensions(int largestIndex, Quaternion value)
```

Comment:
```suggestion
        static Vector3 GetSmallerDimensions(int largestIndex, Quaternion value)
```
uppercase for consistency please